share/picongpu/tests/compileIterationStart/include/picongpu/param/iterationStart.param

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2013-2021 Rene Widera, Richard Pausch, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *
+ * Specify a sequence of functors to be called at start of each time iteration.
+ */
+
+#pragma once
+
+#include "picongpu/particles/InitFunctors.hpp"
+
+
+namespace picongpu
+{
+    //! Functor to move a source particle to the destination species if it has a high gamma value
+    struct MoveIfHighGammaFunctor
+    {
+        /** Process a pair of particles.
+         *
+         * Move a source particle to a destination particle if the source particle has high gamma value.
+         *
+         * @tparam T_DestParticle destination particle type
+         * @tparam T_SrcParticle source particle type
+         *
+         * @param particleDest destination particle
+         * @param particleSrc source particle
+         */
+        template<typename T_DestParticle, typename T_SrcParticle>
+        HDINLINE void operator()(T_DestParticle& particleDest, T_SrcParticle& particleSrc)
+        {
+            auto const gamma = picongpu::gamma<float_X>(
+                particleSrc[picongpu::momentum_],
+                picongpu::traits::attribute::getMass(particleSrc[picongpu::weighting_], particleSrc));
+            constexpr float_X thresholdGamma = 3.0_X;
+            if(gamma >= thresholdGamma)
+            {
+                pmacc::particles::operations::assign(particleDest, particleSrc);
+                particleSrc[picongpu::multiMask_] = 0;
+            }
+            else
+                particleDest[picongpu::multiMask_] = 0;
```

Comment:
Note to me: We need the possibility to create particles only if needed. The current implementation requires the same amount of memory for the destination species as the source species is consuming. The reason is that `ManipulateDerive` which is used to execute this functor is a simple copy where we invalidate a particle if the condition is not fulfilled.